### DISCUSSION TITLE
Create settings

### DISCUSSION THREAD

**Discussion by nguyenphuochien97** [bigcommerce/catalyst#693] [Q&A] [answered]
Hi Team, 

Catalyst has very high performance, I really like this!

In the future, will Catalyst be able to create settings from schema.json files like Cornerstone?

Thank you very much

**Comment by bookernath** [comment, marked as answer]
Thanks for your kind words on the performance :)

We're thinking about the right way to provide connectivity between the Catalyst storefront app and the control panel, so we'd love to have your feedback on things you'd like to see there. My preference would be to expose these things in a very open way using our platform's features, so potentially backing it by metafields for example.

For the parts of "settings" that really just control styles and colors, we hope to move those into the visual editing layer, which can have authority over styling and layout - that's probably 90% of the settings in Cornerstone today.

What do you think? Do you have specific use cases in mind?

**Reply by nguyenphuochien97** [reply]
Thank you so much for your feedback. We greatly appreciate your ideas regarding the Catalyst storefront app and the control panel, and we are excited to see this launched soon.

As theme developers for BigCommerce merchants, when building or setting up a theme layout, we encounter limitations in styling and layout. Therefore, it would be amazing to create a robust visual builder for store/theme layouts.

Based on the current limitations, we have some recommendations:

- Focus on improving the control panel for the theme editor to run faster and smoother during styling and layout adjustments.

- Expand the settings ability for theme layouts so merchants can easily customize or make changes to the layout. When editing the theme in the Customize control panel, we expect to see more visual settings which are easy to use.

- We noticed an issue with data storage on WebDAV, when uploading videos via WebDAV, videos will not work on IOS.

**Reply by bookernath** [reply]
@nguyenphuochien97 you might want to have a look at our Makeswift integration to get a sense of the type of visual editing capabilities we plan to bake into Catalyst in the future - this is an early preview supporting landing pages:

https://docs.makeswift.com/integrations/bigcommerce

**Reply by nguyenphuochien97** [reply]
Wow, that's amazing! The integration between Bigcommerce and Makeswift is remarkable. I'm really looking forward to this new version. It looks like it's going to solve all the previous problems.
Do your team members have an expected release time for it?
